lib/identity_cache/query_api.rb

```diff
@@ -79,7 +79,7 @@ def fetch_multi(*ids)
       def prefetch_associations(associations, records)
         records = records.to_a
         return if records.empty?
-        unless IdentityCache.should_use_cache?
+        unless should_use_cache?
```

Comment:
Elsewhere you seem to be doing this check on each cache_has_many association, but here it is done for multiple cache_has_many associations